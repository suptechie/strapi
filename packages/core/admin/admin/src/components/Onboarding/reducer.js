import produce from 'immer';
import set from 'lodash/set';

const initialState = {
  isLoading: true,
  isOpen: false,
  videos: [],
};

const reducer = (state, action) =>
  // eslint-disable-next-line consistent-return
  produce(state, draftState => {
    switch (action.type) {
      case 'GET_DATA_SUCCEEDED': {
        draftState.isOpen = !action.didWatchVideos;
        draftState.isLoading = false;
        draftState.videos = action.videos;
        break;
      }

      case 'SET_IS_OPEN': {
        draftState.isOpen = !state.isOpen;
        break;
      }
      case 'SET_VIDEO_DURATION': {
        set(draftState, ['videos', action.videoIndex, 'duration'], parseFloat(action.duration, 10));
        break;
      }
      case 'TOGGLE_VIDEO_MODAL': {
        const nextVideos = state.videos.map((video, index) => {
          if (index === action.videoIndexToOpen) {
            return { ...video, isOpen: !video.isOpen };
          }

          return { ...video, isOpen: false };
        });
        draftState.videos = nextVideos;
        break;
      }
      case 'UPDATE_VIDEO_STARTED_TIME_AND_PLAYED_INFOS': {
        const nextVideos = state.videos.map((video, index) => {
          if (index !== action.videoIndex) {
            return video;
          }
          const elapsedTime = parseFloat(action.elapsedTime, 10);
          const videoDuration = parseFloat(video.duration, 10);
          const percentElapsedTime = (elapsedTime * 100) / videoDuration;
          const end = video.end === true ? video.end : percentElapsedTime > 80;

          return { ...video, startTime: elapsedTime, end };
        });

        // Update the local storage and make sure that the modal video does not automatically open
        localStorage.setItem(
          'videos',
          JSON.stringify(nextVideos.map(v => ({ ...v, isOpen: false })))
        );
        // Update the state
        draftState.videos = nextVideos;
        break;
      }
      default:
        return draftState;
    }
  });

export default reducer;
export { initialState };
