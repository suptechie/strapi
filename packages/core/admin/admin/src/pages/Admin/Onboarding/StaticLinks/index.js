/**
 *
 * StaticLinks
 *
 */

import React from 'react';
import { useIntl } from 'react-intl';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Text } from '@strapi/parts/Text';
import { InjectionZone } from '../../../../shared/components';

const StyledLink = styled.a`
  display: flex;
  width: 100%;
  height: 27px;
  padding: 0 20px;
  color: ${({ theme }) => theme.colors.neutral400};
  text-decoration: none;
  span,
  svg {
    margin: auto 0;
  }
  svg {
    margin-right: 10px;
  }

  ${[Text]} {
    color: ${({ theme }) => theme.colors.neutral400};
    margin: auto 0;
  }

  &:hover {
    background: ${({ theme }) => theme.colors.neutral100};
    color: ${({ theme }) => theme.colors.neutral500};

    ${[Text]} {
      color: ${({ theme }) => theme.colors.neutral700};
    }
  }

  ${[Text]} {
    color: ${({ theme }) => theme.colors.neutral600};
  }
`;

const StaticLinks = () => {
  const { formatMessage } = useIntl();
  const staticLinks = [
    {
      icon: 'book',
      label: formatMessage({ id: 'app.components.LeftMenuFooter.documentation' }),
      destination: 'https://strapi.io/documentation',
    },
    {
      icon: 'file',
      label: formatMessage({ id: 'app.static.links.cheatsheet' }),
      destination: 'https://strapi-showcase.s3-us-west-2.amazonaws.com/CheatSheet.pdf',
    },
  ];

  return (
    <ul className="list">
      {staticLinks.map(link => {
        const { icon, label, destination } = link;

        return (
          <li key={label}>
            <StyledLink href={destination} target="_blank" rel="noopener noreferrer">
              <FontAwesomeIcon icon={icon} />
              <Text>{label}</Text>
            </StyledLink>
          </li>
        );
      })}
      <InjectionZone area="admin.tutorials.links" />
    </ul>
  );
};

export default StaticLinks;
