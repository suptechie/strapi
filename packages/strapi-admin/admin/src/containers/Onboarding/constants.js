/*
 *
 * Onboarding constants
 *
 */

export const GET_VIDEOS = 'StrapiAdmin/Onboarding/GET_VIDEOS';
export const GET_VIDEOS_SUCCEEDED =
  'StrapiAdmin/Onboarding/GET_VIDEOS_SUCCEEDED';
  export const ON_CLICK = 'StrapiAdmin/Onboarding/ON_CLICK';
  export const SET_VIDEOS_DURATION = 'StrapiAdmin/Onboarding/SET_VIDEOS_DURATION';
