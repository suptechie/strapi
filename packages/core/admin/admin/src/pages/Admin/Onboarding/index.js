import React, { useRef, useState } from 'react';
import styled from 'styled-components';
import { useIntl } from 'react-intl';
import {
  Box,
  Button,
  Divider,
  Flex,
  FocusTrap,
  Icon,
  Portal,
  PopoverPrimitives,
  Stack,
  Typography,
} from '@strapi/design-system';
import { Cross, Play, Question } from '@strapi/icons';

import { useConfigurations } from '../../../hooks';
import OnboardingPreview from '../../../assets/images/onboarding-preview.png';
import { VIDEO_LINKS, DOCUMENTATION_LINKS } from './constants';

// TODO: use new Button props derived from Box props with next DS release
const HelperButton = styled(Button)`
  border-radius: 50%;
  padding: ${({ theme }) => theme.spaces[3]};
  // Resetting 2rem height defined by Button component
  height: 100%;
`;

const IconWrapper = styled(Flex)`
  transform: translate(-50%, -50%);
`;

const VideoLinkWrapper = styled(Flex)`
  text-decoration: none;

  :focus-visible {
    outline-offset: ${({ theme }) => `-${theme.spaces[1]}`};
  }

  :hover {
    background: ${({ theme }) => theme.colors.primary100};

    // Hover style for the number displayed
    div:first-child span {
      color: ${({ theme }) => theme.colors.primary500};
    }

    // Hover style for the label
    span:nth-child(1) {
      color: ${({ theme }) => theme.colors.primary600};
    }
  }
`;

const Preview = styled.img`
  width: 56px;
  height: 40px;
  // Same overlay used in ModalLayout
  background: ${({ theme }) => `${theme.colors.neutral800}1F`};
  border-radius: ${({ theme }) => theme.borderRadius};
`;

const TextLink = styled(Typography)`
  text-decoration: none;
`;

const Onboarding = () => {
  const buttonRef = useRef();
  const [isOpen, setIsOpen] = useState(false);
  const { formatMessage } = useIntl();
  const { showTutorials } = useConfigurations();

  if (!showTutorials) {
    return null;
  }

  const handlePopoverVisibility = () => {
    setIsOpen((prev) => !prev);
  };

  return (
    <Box as="aside" position="fixed" bottom={2} right={2}>
      <HelperButton
        aria-label={formatMessage(
          isOpen
            ? {
                id: 'app.components.Onboarding.help.button-close',
                defaultMessage: 'Close help menu',
              }
            : {
                id: 'app.components.Onboarding.help.button',
                defaultMessage: 'Open help menu',
              }
        )}
        onClick={handlePopoverVisibility}
        ref={buttonRef}
      >
        <Icon as={isOpen ? Cross : Question} color="buttonNeutral0" />
      </HelperButton>

      {isOpen && (
        <Portal>
          <PopoverPrimitives.Content
            padding={0}
            source={buttonRef}
            placement="top-end"
            spacing={12}
          >
            <FocusTrap onEscape={handlePopoverVisibility}>
              <Flex
                alignItems="end"
                justifyContent="space-between"
                paddingBottom={4}
                paddingRight={5}
                paddingLeft={5}
                paddingTop={5}
              >
                <Typography fontWeight="bold">
                  {formatMessage({
                    id: 'app.components.Onboarding.title',
                    defaultMessage: 'Get started videos',
                  })}
                </Typography>
                <TextLink
                  as="a"
                  href="https://www.youtube.com/playlist?list=PL7Q0DQYATmvidz6lEmwE5nIcOAYagxWqq"
                  target="_blank"
                  rel="noreferrer noopener"
                  variant="pi"
                  textColor="primary600"
                >
                  {formatMessage({
                    id: 'app.components.Onboarding.link.more-videos',
                    defaultMessage: 'Watch more videos',
                  })}
                </TextLink>
              </Flex>
              <Divider />
              {VIDEO_LINKS.map(({ href, duration, label }, index) => (
                <VideoLinkWrapper
                  as="a"
                  href={href}
                  target="_blank"
                  rel="noreferrer noopener"
                  key={href}
                  hasRadius
                  paddingTop={4}
                  paddingBottom={4}
                  paddingLeft={6}
                  paddingRight={11}
                >
                  <Box paddingRight={5}>
                    <Typography textColor="neutral200" variant="alpha">
                      {index + 1}
                    </Typography>
                  </Box>
                  <Box position="relative">
                    <Preview src={OnboardingPreview} alt="onboarding preview" />
                    <IconWrapper
                      position="absolute"
                      top="50%"
                      left="50%"
                      background="primary600"
                      borderRadius="50%"
                      justifyContent="center"
                      width={6}
                      height={6}
                    >
                      <Icon as={Play} color="buttonNeutral0" width={3} height={3} />
                    </IconWrapper>
                  </Box>
                  <Flex direction="column" alignItems="start" paddingLeft={4}>
                    <Typography fontWeight="bold">{formatMessage(label)}</Typography>
                    <Typography textColor="neutral600" variant="pi">
                      {duration}
                    </Typography>
                  </Flex>
                </VideoLinkWrapper>
              ))}
              <Stack spacing={2} paddingLeft={5} paddingTop={2} paddingBottom={5}>
                {DOCUMENTATION_LINKS.map(({ label, href, icon }) => (
                  <Stack horizontal spacing={3} key={href}>
                    <Icon as={icon} color="primary600" />
                    <TextLink
                      as="a"
                      href={href}
                      target="_blank"
                      rel="noreferrer noopener"
                      variant="sigma"
                      textColor="primary700"
                    >
                      {formatMessage(label)}
                    </TextLink>
                  </Stack>
                ))}
              </Stack>
            </FocusTrap>
          </PopoverPrimitives.Content>
        </Portal>
      )}
    </Box>
  );
};

export default Onboarding;
