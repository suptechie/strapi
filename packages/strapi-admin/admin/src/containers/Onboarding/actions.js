/*
 *
 * Onboarding actions
 *
 */

import { GET_VIDEOS, GET_VIDEOS_SUCCEEDED, ON_CLICK } from './constants';

export function getVideos() {
  return {
    type: GET_VIDEOS,
  };
}

export function getVideosSucceeded(videos) {
  return {
    type: GET_VIDEOS_SUCCEEDED,
    videos,
  };
}
export function onClick(e) {
  return {
    type: ON_CLICK,
    index: parseInt(e.currentTarget.id, 10),
  };
}
