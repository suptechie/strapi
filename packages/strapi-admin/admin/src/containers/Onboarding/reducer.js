/*
 *
 * Onboarding reducer
 *
 */

import { fromJS } from 'immutable';
import { GET_VIDEOS_SUCCEEDED, ON_CLICK, SET_VIDEOS_DURATION, UPDATE_VIDEO_START_TIME } from './constants';

const initialState = fromJS({
  videos: fromJS([]),
});

function onboardingReducer(state = initialState, action) {
  switch (action.type) {
    case GET_VIDEOS_SUCCEEDED:
      return state.update('videos', () => fromJS(action.videos));
    case ON_CLICK:
      return state.updateIn(['videos'], list => {
        return list.reduce((acc, current, index) => {

          if (index === action.index) {
            return acc.updateIn([index, 'isOpen'], v => !v);
          }

          return acc.updateIn([index, 'isOpen'], () => false);
        }, list);
      });
    case SET_VIDEOS_DURATION:
      return state.updateIn(['videos', action.index, 'duration'], () => action.duration);
    case UPDATE_VIDEO_START_TIME:
      return state.updateIn(['videos'], list => {
        return list.reduce((acc, current, index) => {

          if (index === action.index) {
            return acc.updateIn([index, 'startTime'], () => action.startTime);
          }

          return acc.updateIn([index, 'startTime'], () => 0);
        }, list);
      });
    default:
      return state;
  }
}

export default onboardingReducer;
