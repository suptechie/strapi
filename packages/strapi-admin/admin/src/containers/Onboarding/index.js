/**
 *
 * Onboarding
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { connect } from 'react-redux';
import { bindActionCreators, compose } from 'redux';
import { getVideos, onClick, setVideoDuration, updateVideoStartTime, setVideoEnd, removeVideos } from './actions';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import makeSelectOnboarding from './selectors';
import reducer from './reducer';
import saga from './saga';

import OnboardingVideo from 'components/OnboardingVideo';

import styles from './styles.scss';

export class Onboarding extends React.Component {
  state = { showVideos: false };

  componentDidMount() {
    this.setState({ showVideos: true });
    this.props.getVideos();
  }
  
  componentWillUnmount() {
    this.props.removeVideos();
    localStorage.removeItem('videos');
  }

  toggleVideos = () => {
    // Display videos card
    this.setState(prevState => ({ showVideos: !prevState.showVideos }));

    // EmitEvent
    const { showVideos } = this.state;
    const eventName = showVideos ? 'didOpenGetStartedVideoContainer' : 'didCloseGetStartedVideoContainer';
    this.context.emitEvent(eventName);
  };

  setVideoEnd = () => {
    this.setVideoEnd();
  }

  didPlayVideo = (index, currTime) => {

    const eventName = `didPlay${index}GetStartedVideo`;
    this.context.emitEvent(eventName, {timestamp: currTime});
  }

  didStopVideo = (index, currTime) => {

    const eventName = `didStop${index}Video`;
    this.context.emitEvent(eventName, {timestamp: currTime});
  }

  updateLocalStorage = (index, current, duration) => {
    // Update store
    this.props.updateVideoStartTime(index, current);

    // Update localStorage
    let videosTime = JSON.parse(localStorage.getItem('videos'));
    videosTime[index].startTime = current;

    let percent = current * 100 / duration;
    if (percent >= 80) {
      if (videosTime[index].end === false) {
        videosTime[index].end = true;
        this.props.setVideoEnd(index, true);
      }
    }

    localStorage.setItem('videos', JSON.stringify(videosTime));
  };

  // eslint-disable-line react/prefer-stateless-function
  render() {
    const { videos, onClick, setVideoDuration } = this.props;

    return (
      <div className={styles.videosWrapper}>
        <div
          className={cn(
            styles.videosContent,
            this.state.showVideos ? styles.shown : styles.hide,
          )}
        >
          <div className={styles.videosHeader}>
            <p>Get started video</p>
            <p>25% completed</p>
          </div>

          <ul className={styles.onboardingList}>
            {videos.map((video, i) => {
              return (
                <OnboardingVideo
                  key={i}
                  id={i}
                  video={video}
                  onClick={onClick}
                  setVideoDuration={setVideoDuration}
                  getVideoCurrentTime={this.updateLocalStorage}
                  didPlayVideo={this.didPlayVideo}
                  didStopVideo={this.didStopVideo}
                />
              );
            })}
          </ul>
        </div>

        <div className={styles.openBtn}>
          <button
            onClick={this.toggleVideos}
            className={this.state.showVideos ? styles.active : ''}
          >
            <i className="fa fa-question" />
            <i className="fa fa-times" />
            <span />
          </button>
        </div>
      </div>
    );
  }
}
Onboarding.contextTypes = {
  emitEvent: PropTypes.func,
};

Onboarding.propTypes = {
  getVideos: PropTypes.func.isRequired,
};

const mapStateToProps = makeSelectOnboarding();

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ getVideos, onClick, setVideoDuration, updateVideoStartTime, setVideoEnd, removeVideos }, dispatch);
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

/* Remove this line if the container doesn't have a route and
 *  check the documentation to see how to create the container's store
 */
const withReducer = injectReducer({ key: 'onboarding', reducer });

/* Remove the line below the container doesn't have a route and
 *  check the documentation to see how to create the container's store
 */
const withSaga = injectSaga({ key: 'onboarding', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(Onboarding);
