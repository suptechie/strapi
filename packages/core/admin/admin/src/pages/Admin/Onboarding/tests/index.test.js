import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { ThemeProvider, lightTheme } from '@strapi/design-system';
import Onboarding from '../index';

jest.mock('../../../../hooks', () => ({
  useConfigurations: jest.fn(() => {
    return { showTutorials: true };
  }),
}));

const App = (
  <ThemeProvider theme={lightTheme}>
    <IntlProvider locale="en" messages={{}} defaultLocale="en" textComponent="span">
      <Onboarding />
    </IntlProvider>
  </ThemeProvider>
);

describe('Onboarding', () => {
  it('renders and matches the snapshot', async () => {
    const {
      container: { firstChild },
    } = render(App);

    expect(firstChild).toMatchInlineSnapshot(`
      .c0 {
        position: fixed;
        bottom: 8px;
        right: 8px;
      }

      .c1 {
        width: 40px;
        height: 40px;
        background: #4945ff;
        box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
        border-radius: 50%;
      }

      .c1 svg path {
        fill: #ffffff;
      }

      <aside
        class="c0"
      >
        <button
          aria-label="Help button"
          class="c1"
          id="onboarding"
        >
          <svg
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 0C5.383 0 0 5.383 0 12s5.383 12 12 12 12-5.383 12-12S18.617 0 12 0zm0 4.92a1.56 1.56 0 110 3.12 1.56 1.56 0 010-3.12zm3.84 13.06a.5.5 0 01-.5.5h-6.2a.5.5 0 01-.5-.5v-.92a.5.5 0 01.5-.5h2.14v-5.28H9.86a.5.5 0 01-.5-.5v-.92a.5.5 0 01.5-.5h2.84a.5.5 0 01.5.5v6.7h2.14a.5.5 0 01.5.5v.92z"
              fill="#212134"
            />
          </svg>
        </button>
      </aside>
    `);
  });

  it('should open links when button is clicked', () => {
    render(App);

    fireEvent.click(document.querySelector('#onboarding'));
    expect(screen.getByText('Documentation')).toBeInTheDocument();
  });
});
