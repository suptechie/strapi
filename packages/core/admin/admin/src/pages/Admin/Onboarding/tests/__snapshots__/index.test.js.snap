// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding renders and matches the snapshot 1`] = `
.c6 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #32324d;
}

.c14 {
  font-size: 0.875rem;
  line-height: 1.43;
  font-weight: 600;
  color: #32324d;
}

.c15 {
  font-size: 0.75rem;
  line-height: 1.33;
  color: #4945ff;
}

.c23 {
  font-weight: 600;
  font-size: 2rem;
  line-height: 1.25;
  color: #dcdce4;
}

.c32 {
  font-size: 0.75rem;
  line-height: 1.33;
  color: #666687;
}

.c39 {
  font-weight: 600;
  font-size: 0.6875rem;
  line-height: 1.45;
  text-transform: uppercase;
  color: #271fe0;
}

.c1 {
  position: fixed;
  right: 8px;
  bottom: 8px;
}

.c7 {
  color: #ffffff;
}

.c10 {
  background: #ffffff;
  border-radius: 4px;
}

.c12 {
  padding-top: 20px;
  padding-right: 20px;
  padding-bottom: 16px;
  padding-left: 20px;
}

.c17 {
  background: #eaeaef;
}

.c19 {
  padding-top: 16px;
  padding-right: 64px;
  padding-bottom: 16px;
  padding-left: 24px;
  border-radius: 4px;
}

.c22 {
  padding-right: 20px;
}

.c24 {
  position: relative;
}

.c26 {
  background: #4945ff;
  border-radius: 50%;
  position: absolute;
  left: 50%;
  top: 50%;
  width: 24px;
  height: 24px;
}

.c29 {
  color: #ffffff;
  width: 12px;
  height: 12px;
}

.c30 {
  padding-left: 16px;
}

.c33 {
  padding-top: 8px;
  padding-bottom: 20px;
  padding-left: 20px;
}

.c37 {
  color: #4945ff;
}

.c13 {
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c20 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c27 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c31 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c34 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c35 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c35 > * + * {
  margin-top: 8px;
}

.c36 > * {
  margin-left: 0;
  margin-right: 0;
}

.c36 > * + * {
  margin-left: 12px;
}

.c8 path {
  fill: #ffffff;
}

.c38 path {
  fill: #4945ff;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
  position: relative;
  outline: none;
}

.c2 svg {
  height: 12px;
  width: 12px;
}

.c2 svg > g,
.c2 svg path {
  fill: #ffffff;
}

.c2[aria-disabled='true'] {
  pointer-events: none;
}

.c2:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c2:focus-visible {
  outline: none;
}

.c2:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c9 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4945ff;
  border: 1px solid #4945ff;
  height: 2rem;
  padding-left: 16px;
  padding-right: 16px;
}

.c3 .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 .c5 {
  color: #ffffff;
}

.c3[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c3[aria-disabled='true'] .c5 {
  color: #666687;
}

.c3[aria-disabled='true'] svg > g,.c3[aria-disabled='true'] svg path {
  fill: #666687;
}

.c3[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c3[aria-disabled='true']:active .c5 {
  color: #666687;
}

.c3[aria-disabled='true']:active svg > g,.c3[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c3:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c3:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c3 svg > g,
.c3 svg path {
  fill: #ffffff;
}

.c11 {
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
  z-index: 4;
  border: 1px solid #eaeaef;
  background: #ffffff;
}

.c18 {
  height: 1px;
  border: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin: 0;
}

.c4 {
  border-radius: 50%;
  padding: 12px;
  height: 100%;
}

.c28 {
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c21 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c21:focus-visible {
  outline-offset: -4px;
}

.c21:hover {
  background: #f0f0ff;
}

.c21:hover div:first-child span {
  color: #7b79ff;
}

.c21:hover span:nth-child(1) {
  color: #4945ff;
}

.c25 {
  width: 56px;
  height: 40px;
  background: #32324d1F;
  border-radius: 4px;
}

.c16 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<body>
  <div>
    <aside
      class="c0 c1"
    >
      <button
        aria-disabled="false"
        aria-label="Close help menu"
        class="c2 c3 c4"
        type="button"
      >
        <span
          class="c5 c6"
        >
          <svg
            class="c0 c7 c8"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 2.417L21.583 0 12 9.583 2.417 0 0 2.417 9.583 12 0 21.583 2.417 24 12 14.417 21.583 24 24 21.583 14.417 12 24 2.417z"
              fill="#212134"
            />
          </svg>
        </span>
      </button>
    </aside>
    <div
      class="c9"
    >
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-log"
        role="log"
      />
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-status"
        role="status"
      />
      <p
        aria-live="assertive"
        aria-relevant="all"
        id="live-region-alert"
        role="alert"
      />
    </div>
  </div>
  <div
    data-react-portal="true"
  >
    <div
      class="c0 c10 c11"
      style="position: fixed;"
    >
      <div>
        <div
          class="c0 c12 c13"
        >
          <span
            class="c5 c14"
          >
            Get started videos
          </span>
          <a
            class="c5 c15 c16"
            href="https://www.youtube.com/playlist?list=PL7Q0DQYATmvidz6lEmwE5nIcOAYagxWqq"
            rel="noreferrer noopener"
            target="_blank"
          >
            Watch more videos
          </a>
        </div>
        <hr
          class="c0 c17 c18"
        />
        <a
          class="c0 c19 c20 c21"
          href="https://www.youtube.com/watch?v=G9GjN0RxhkE"
          rel="noreferrer noopener"
          target="_blank"
        >
          <div
            class="c0 c22"
          >
            <span
              class="c5 c23"
            >
              1
            </span>
          </div>
          <div
            class="c0 c24"
          >
            <img
              alt="onboarding preview"
              class="c25"
              src="IMAGE_MOCK"
            />
            <div
              class="c0 c26 c27 c28"
            >
              <svg
                class="c0 c29 c8"
                fill="none"
                height="1em"
                viewBox="0 0 8 8"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2 .93c0-.4.45-.63.78-.41l4.6 3.06c.3.2.3.64 0 .84l-4.6 3.06A.5.5 0 012 7.07V.93z"
                  fill="#212134"
                />
              </svg>
            </div>
          </div>
          <div
            class="c0 c30 c31"
          >
            <span
              class="c5 c14"
            >
              Build a content architecture
            </span>
            <span
              class="c5 c32"
            >
              5:48
            </span>
          </div>
        </a>
        <a
          class="c0 c19 c20 c21"
          href="https://www.youtube.com/watch?v=DEZw4KbybAI"
          rel="noreferrer noopener"
          target="_blank"
        >
          <div
            class="c0 c22"
          >
            <span
              class="c5 c23"
            >
              2
            </span>
          </div>
          <div
            class="c0 c24"
          >
            <img
              alt="onboarding preview"
              class="c25"
              src="IMAGE_MOCK"
            />
            <div
              class="c0 c26 c27 c28"
            >
              <svg
                class="c0 c29 c8"
                fill="none"
                height="1em"
                viewBox="0 0 8 8"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2 .93c0-.4.45-.63.78-.41l4.6 3.06c.3.2.3.64 0 .84l-4.6 3.06A.5.5 0 012 7.07V.93z"
                  fill="#212134"
                />
              </svg>
            </div>
          </div>
          <div
            class="c0 c30 c31"
          >
            <span
              class="c5 c14"
            >
              Add & manage content
            </span>
            <span
              class="c5 c32"
            >
              3:18
            </span>
          </div>
        </a>
        <a
          class="c0 c19 c20 c21"
          href="https://www.youtube.com/watch?v=-61MuiMQb38"
          rel="noreferrer noopener"
          target="_blank"
        >
          <div
            class="c0 c22"
          >
            <span
              class="c5 c23"
            >
              3
            </span>
          </div>
          <div
            class="c0 c24"
          >
            <img
              alt="onboarding preview"
              class="c25"
              src="IMAGE_MOCK"
            />
            <div
              class="c0 c26 c27 c28"
            >
              <svg
                class="c0 c29 c8"
                fill="none"
                height="1em"
                viewBox="0 0 8 8"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2 .93c0-.4.45-.63.78-.41l4.6 3.06c.3.2.3.64 0 .84l-4.6 3.06A.5.5 0 012 7.07V.93z"
                  fill="#212134"
                />
              </svg>
            </div>
          </div>
          <div
            class="c0 c30 c31"
          >
            <span
              class="c5 c14"
            >
              Manage media
            </span>
            <span
              class="c5 c32"
            >
              3:41
            </span>
          </div>
        </a>
        <div
          class="c0 c33 c34 c35"
        >
          <div
            class="c0 c20 c36"
          >
            <svg
              class="c0 c37 c38"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13 21v2.5l-3-2-3 2V21h-.5A3.5 3.5 0 013 17.5V5a3 3 0 013-3h14a1 1 0 011 1v17a1 1 0 01-1 1h-7zm-6-2v-2h6v2h6v-3H6.5a1.5 1.5 0 100 3H7zM7 5v2h2V5H7zm0 3v2h2V8H7zm0 3v2h2v-2H7z"
                fill="#212134"
              />
            </svg>
            <a
              class="c5 c39 c16"
              href="https://docs.strapi.io"
              rel="noreferrer noopener"
              target="_blank"
            >
              documentation
            </a>
          </div>
          <div
            class="c0 c20 c36"
          >
            <svg
              class="c0 c37 c38"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.946 9.315c-.522-.174-.527-.455.01-.634l19.087-6.362c.53-.176.832.12.684.638l-5.454 19.086c-.15.529-.455.547-.679.045L12 14l6-8-8 6-8.054-2.685z"
                fill="#212134"
              />
            </svg>
            <a
              class="c5 c39 c16"
              href="https://strapi-showcase.s3-us-west-2.amazonaws.com/CheatSheet.pdf"
              rel="noreferrer noopener"
              target="_blank"
            >
              cheatsheet
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
